cmd/buck/cli/archive.go

```diff
@@ -63,47 +72,71 @@ var setDefaultArchiveConfigCmd = &cobra.Command{
 			buf := new(bytes.Buffer)
 			_, err = buf.ReadFrom(reader)
 			cmd.ErrCheck(err)
+
 			cmd.ErrCheck(json.Unmarshal(buf.Bytes(), &config))
 		} else {
-			repFactor, err := c.Flags().GetInt("rep-factor")
+			ctx, cancel := context.WithTimeout(context.Background(), cmd.Timeout)
+			defer cancel()
+			buck, err := bucks.GetLocalBucket(ctx, conf)
 			cmd.ErrCheck(err)
-			config.RepFactor = repFactor
-
-			dealMinDuration, err := c.Flags().GetInt64("deal-min-duration")
+			config, err = buck.DefaultArchiveConfig(ctx)
```

Comment:
We assign to `config` the current default storage-config, then we mutate if flags are explicitly set.